Sources/swift-doc/Supporting Types/Components/Relationships.swift

```diff
@@ -73,30 +73,36 @@ struct Relationships: Component {
     }
 
     var html: HypertextLiteral.HTML {
-        var graph = symbol.graph(in: module)
-        guard !graph.edges.isEmpty else { return "" }
-
-        graph.aspectRatio = 0.125
-        graph.center = true
-        graph.overlap = "compress"
+        guard sections.map(\.symbols.count).reduce(0, +) != 0 else { return "" }
```

Comment:
I'm still on the fence about Swift's new keypath `map` syntax. But setting that aside: We don't need to compute the total count to determine if there are any symbols. A better way would be to do `let _ = sections.first { !$0.symbols.isEmpty }`.

Reply:
I'm not sure about checking the first item only:
- I noticed that `sections` are processed with `filter { !$0.symbols.isEmpty }` so there should be no empty sections at the moment, we can just test for `sections.isEmpty`
- at the same time, I see `guard !symbols.isEmpty else { return nil }` in the generation code

So I decided to write a more defensive check to be sure that it won't break in the future.

I'd prefer to state that we never have empty sections and simplify/remove all relevant checks.

Reply:
To clarify: the use of `first` doesn't check the first item, but returns as soon as it finds a section with non-empty `symbols`. But taking another look, you're right: a simple check on `sections.isEmpty` should be sufficient. The other guard statement you mentioned was from an earlier iteration, and I neglected to remove it.

Reply:
Oh, right, I've misunderstood your initial suggestion. Thanks for clarifying that.

Changing to `isEmpty` + removed that `guard` statement.